firmware/thruna/DCM/Src/App/App_Regen.c

```diff
@@ -0,0 +1,144 @@
+#include "App_Regen.h"
+#include <stdlib.h>
+#include "math.h"
+
+/**
+ * Check if left or right wheel is greater than 5.0km/hr
+ * @return true if wheel speed meets this condition, false
+ * otherwise
+ */
+static bool wheel_speed_in_range();
+
+/**
+ * Check if battery cells are less than 4.0V
+ * @param RegenBraking struct to populate data
+ * @return true battery cells meet this condition,
+ * false otherwise
+ */
+static bool power_limit_check(RegenBraking *regenAttr);
+
+/**
+ * Algorithm to send negative torque request dependent
+ * on accelerator pedal percentage in range [-100, 0]
+ * and to do active differential or not
+ * @param inputs are inputs for active differential
+ * @param regenAttr struct provides for torque request
+ */
+static void compute_regen_torque_request(ActiveDifferential_Inputs *inputs, RegenBraking *regenAttr);
+
+static RegenBraking              regenAttributes = { .enable_active_differential = true };
+static ActiveDifferential_Inputs activeDifferentialInputs;
+
+void App_Run_Regen(float *prev_torque_request, float accelerator_pedal_percentage)
+{
+    activeDifferentialInputs.accelerator_pedal_percentage = accelerator_pedal_percentage;
+
+    if (App_Regen_SafetyCheck(&regenAttributes))
+    {
+        regenAttributes.prev_torque_request_Nm = *prev_torque_request;
+        activeDifferentialInputs.steering_angle_deg =
+            App_CanRx_FSM_SteeringAngle_Get() * APPROX_STEERING_TO_WHEEL_ANGLE;
+
+        compute_regen_torque_request(&activeDifferentialInputs, &regenAttributes);
+    }
+    else
+    {
+        regenAttributes.left_inverter_torque_Nm  = 0.0;
+        regenAttributes.right_inverter_torque_Nm = 0.0;
+        regenAttributes.prev_torque_request_Nm   = 0.0;
+
+        App_CanTx_DCM_Warning_RegenNotAvailable_Set(true);
+    }
+
+    *prev_torque_request = regenAttributes.prev_torque_request_Nm;
+
+    App_Regen_Send_Torque_Request(regenAttributes.left_inverter_torque_Nm, regenAttributes.right_inverter_torque_Nm);
```

Comment:
can you return this, and pass it into `App_SetPeriodicCanSignals_TorqueRequests`